pkg/scheduler/framework/plugins/interpodaffinity/plugin.go

```diff
@@ -71,3 +76,37 @@ func getArgs(obj runtime.Object) (config.InterPodAffinityArgs, error) {
 	}
 	return *ptr, nil
 }
+
+// Updates Namespaces with the set of namespaces identified by NamespaceSelector.
+// If successful, NamespaceSelector is set to nil.
+// The assumption is that the term is for an incoming pod, in which case
+// namespaceSelector is either unrolled into Namespaces (and so the selector
+// is set to Nothing()) or is Empty(), which means match everything. Therefore,
+// there when matching against this term, there is no need to lookup the existing
+// pod's namespace labels to match them against term's namespaceSelector explicitly.
+func mergeAffinityTermNamespacesIfNotEmpty(at *framework.AffinityTerm, nsLister listersv1.NamespaceLister) error {
+	if at.NamespaceSelector.Empty() {
+		return nil
+	}
+	ns, err := nsLister.List(at.NamespaceSelector)
```

Comment:
I am skeptical that those optimizations will save us much in practice because a single existing pod is enough to tigger the unrolling... Of course we can come up with a benchmark that will make use of those optimizations, but I don't think this will be common or realistic.